plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearchBackpressure.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testcontainers.containers.Network;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static io.trino.plugin.elasticsearch.ElasticsearchQueryRunner.createElasticsearchQueryRunner;
+import static io.trino.tpch.TpchTable.ORDERS;
+
+public class TestElasticsearchBackpressure
+        extends AbstractTestQueryFramework
+{
+    private static final String image = "elasticsearch:7.0.0";
+
+    private ElasticsearchServer elasticsearch;
+    private ElasticsearchProxy elasticsearchProxy;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Network network = Network.newNetwork();
+        elasticsearch = new ElasticsearchServer(network, image, ImmutableMap.of());
+        elasticsearchProxy = new ElasticsearchProxy(network, 1);
+
+        return createElasticsearchQueryRunner(
+                elasticsearchProxy.getAddress(),
+                ImmutableList.of(ORDERS),
+                ImmutableMap.of(),
+                ImmutableMap.of("elasticsearch.min-retry-time", "1s", "elasticsearch.max-retry-time", "5m"));
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+            throws IOException
+    {
+        elasticsearchProxy.stop();
+        elasticsearch.stop();
+    }
+
+    @Test
```

Comment:
would it be possible to actually trigger retry and ensure jmx stats (alltime count and delay) are > 0?